database.go

```diff
@@ -0,0 +1,16 @@
+package main
+
+type Database interface {
+	saveDevice(device Device) (Device, error)
+}
+
+type InMemoryDatabase struct {
+	devices []Device
+}
+
+func (db *InMemoryDatabase) saveDevice(device Device) (Device, error) {
+	device.Id = len(db.devices)
+	db.devices = append(db.devices, device)
```

Comment:
Since you did this task so fast 😃 I have a question: what if someone tries to add multiple devices at the same time? will this code behave well?